dot/state/block.go

```diff
@@ -250,14 +250,64 @@ func (bs *BlockState) GetHashByNumber(num uint) (common.Hash, error) {
 	return common.NewHash(bh), nil
 }
 
+// GetAllDescendants gets all the descendants for a given block hash, by first checking in memory
+// and, if not found, reading from the block state database.
+func (bs *BlockState) GetAllDescendants(hash common.Hash) ([]common.Hash, error) {
+	allDescendants, err := bs.bt.GetAllDescendants(hash)
+	if err != nil && !errors.Is(err, blocktree.ErrNodeNotFound) {
+		return nil, err
+	}
+
+	if err == nil {
+		return allDescendants, nil
+	}
+
+	// adding itself as descendant to keep this method similar to blocktree.GetAllDescendants
+	allDescendants = []common.Hash{hash}
+
+	header, err := bs.GetHeader(hash)
+	if err != nil {
+		return nil, fmt.Errorf("getting header: %w", err)
+	}
+
+	// TODO: Use GetBlocksByNumber after https://github.com/ChainSafe/gossamer/issues/2748 is done
+	nextBlock, err := bs.GetBlockByNumber(header.Number + 1)
+	if err != nil {
+		return nil, fmt.Errorf("getting block by number: %w", err)
+	}
```

Comment:
I feel like this isn't fork aware? I think we should implement `GetBlocksByNumberAndParent(num uint, parent common.Hash) ([]types.Block, error)`.  This code would be a little cleaner, and we don't have to do the parent hash check on line 280.

Reply:
yes, we should that's why I have created this issue https://github.com/ChainSafe/gossamer/issues/2748. Do you want to wait for `GetBlocksByNumberAndParent` to get merged before this PR gets into development?

Reply:
Yea I think that's better.  I'd rather not risk committing code that isn't fork aware and forget about addressing it.

Reply:
https://github.com/ChainSafe/gossamer/pull/3097